### DISCUSSION TITLE
php-fpm: pool www

### DISCUSSION THREAD

**Discussion by dbrianhendrix** [mcguirepr89/BirdNET-Pi#520] [Q&A]
My BirdNET-Pi has been unreachable for a day now. The only symptom showing was the network ACT light was constantly blinking at a high rate. I finally had time to work on it without success. I finally just powered it off, removed the SD card, reinserted it, and powered back up. It is now actually up and I can SSH into it. It is now at a VERY HIGH load with numerous processes using up CPU -- php-fpm: pool www.
<img width="1108" alt="Screen Shot 2022-08-26 at 7 33 26 AM" src="https://user-images.githubusercontent.com/52499277/186905216-511b1eb0-881e-4c7c-b934-7e9fd0b5f1de.png">

**Comment by mcguirepr89** [comment]
Hi, @dbrianhendrix 

I have no idea what PHP is trying to do. Let's look at a log.

```
journalctl --boot --no-pager -uphp7.4-fpm
```

**Comment by dbrianhendrix** [comment]
```
***@***.***:~# journalctl --boot --no-pager -uphp7.4-fpm
-- Journal begins at Sun 2022-07-31 07:10:01 CDT, ends at Fri 2022-08-26 07:59:45 CDT. --Aug 25 13:46:46 birdnetpi systemd[1]: Starting The PHP 7.4 FastCGI Process Manager...Aug 25 13:46:48 birdnetpi systemd[1]: Started The PHP 7.4 FastCGI Process Manager.
```

**Reply by mcguirepr89** [reply]
I don't see anything unusual there. Do you still see those processes?

**Comment by dbrianhendrix** [comment]
They are still there but they are no longer maxing out the CPU.

root@birdnetpi:~# ps -ef | grep php-fpm
root         533       1  0 07:25 ?        00:00:01 php-fpm: master process (/etc/php/7.4/fpm/php-fpm.conf)
caddy      20968     533  4 07:57 ?        00:11:49 php-fpm: pool www
caddy      70946     533  4 09:13 ?        00:08:14 php-fpm: pool www
caddy     125289     533  4 10:37 ?        00:04:04 php-fpm: pool www
root      184620  183862  0 12:08 pts/0    00:00:00 grep php-fpm

Rebooting did not fix the issue. I had to unplug from power, pull out the SD card, reinsert it, and power on.

**Reply by mcguirepr89** [reply]
To be clear, when you say "Rebooting did not fix the issue. I had to unplug from power, pull out the SD card, reinsert it, and power on." . . . "They are still there but they are no longer maxing out the CPU."

I'm afraid I'm unsure on the issue -- are the processes causing trouble? It is perfectly normal for there to be multiple php processes running on a server like BirdNET-Pi -- there are lots of things in the web interface that are listening and polling the database and reacting to web interface interaction.

It is also normal for the CPU's load to be high since this model processing is resource intensive. (Indeed, none of the php-fpm pool processes were using more than 71% CPU, so I wouldn't say that is cause for alarm.)

I suppose I would actually want to see a problem before trying to address things that we believe are symptoms of a problem that's not yet been identified.

Let me know what else you are seeing that is causing alarm and making you feel you need to power down, remove the SD card, replace the SD card (not sure what those steps do, since RAM will wait for capacitors to discharge if it were a memory thing?) and reboot just to address what could (based on the info we have seen so far) equally likely be explained as a networking issue.

**Comment by dbrianhendrix** [comment]
body{font-family:Helvetica,Arial;font-size:13px}The Raspberry Pi stopped being accessible. It couldn’t be pinged, SSH didn’t work, and the network ACT light was constantly flashing. Nothing I did could “reset” it. Reboots just stayed inaccessible. However, the final time, I unplugged power and then removed the SD card and checked it on my Mac. Nothing was corrupt or missing. However, when I put the SD card back in and powered up, I could finally get to it but it was EXTREMELY slow. The CPU was maxed out (4 at 98.7%) with those processes. Maybe it was playing catch-up. It took over an hour before the system settled down and now appears to be okay.  D. Brian Hendrixveteran  |  mac guy  |  birder  |  coffee snobTennessee Naturalist, firstSenior Linux Engineer, ***@***.*** On August 26, 2022 at 1:30:34 PM, Patrick McGuire ***@***.***) wrote: 
To be clear, when you say "Rebooting did not fix the issue. I had to unplug from power, pull out the SD card, reinsert it, and power on." . . . "They are still there but they are no longer maxing out the CPU."
I'm afraid I'm unsure on the issue -- are the processes causing trouble? It is perfectly normal for there to be multiple php processes running on a server like BirdNET-Pi -- there are lots of things in the web interface that are listening and polling the database and reacting to web interface interaction.
It is also normal for the CPU's load to be high since this model processing is resource intensive. (Indeed, none of the php-fpm pool processes were using more than 71% CPU, so I wouldn't say that is cause for alarm.)
I suppose I would actually want to see a problem before trying to address things that we believe are symptoms of a problem that's not yet been identified.
Let me know what else you are seeing that is causing alarm and making you feel you need to power down, remove the SD card, replace the SD card (not sure what those steps do, since RAM will wait for capacitors to discharge if it was a memory thing?) and reboot just to address what could (based on the info we have seen so far) equally likely to be a networking issue.

**Reply by mcguirepr89** [reply]
Then it sounds like we're going to want a lot more logs for the next time this happens.

Next time that the Pi becomes unresponsive and you need to reboot (and any other measures), we'll look through logs since that will really be the only thing that can tell us one way or another what is going on.

In that case, please run the following command the next time you gain access to the shell:
```
journalctl -b-1 --no-pager -e | tail -n 500 > ~/whyiwentoffline.log
```
and then send that this way?
